### PR TITLE
meson CI: add FreeBSD and MSYS

### LINKED ISSUE #2104: Add FreeBSD to CI
Hi,

Not sure if this is categorized correctly but it would be appreciated if you could at least add one BSD-distro to CI such as FreeBSD (I know MacOS is kinda BSD-ish though).

I'd also like to thank everyone for their work and effort on exiv2.

Best regards,
Daniel

### PR DESCRIPTION
**Pull request by neheb**
Fixes: https://github.com/Exiv2/exiv2/issues/2104

### CODE REVIEW

File: .github/workflows/on_PR_meson_clang.yaml [hunk 1]
```diff
@@ -9,12 +9,11 @@ concurrency:
 jobs:
   Ubuntu:
     runs-on: ubuntu-20.04
-    name: Linux-Clang${{matrix.cxx}}-${{matrix.deps}}-${{matrix.type}}
+    name: Linux-Clang${{matrix.cxx}}-${{matrix.deps}}
     strategy:
       matrix:
         cxx: ['7', '8', '9', '10', '12']
```
Comment (kmilos) [inline .github/workflows/on_PR_meson_clang.yaml:15]: Let's bump this to ubuntu-22.04 and some newer LLVM versions?

Comment (kmilos) [reply .github/workflows/on_PR_meson_clang.yaml:15]: And same for the GCC matrix as well, I don't see the point in checking/supporting this far back...

Comment (neheb) [reply .github/workflows/on_PR_meson_clang.yaml:15]: If it's possible without too much headache, why not?

Newer compilers are tested with other CI.

**Comment by neheb** [comment]
Modified to build on both FreeBSD and MSYS2. Fixed bugs there.

**Comment by kmilos** [comment]
Remove geotag and iconv changes as they are separate PRs now?

File: .github/workflows/on_PR_meson_clang.yaml (hunk 1 above)
Comment (kmilos) [reply .github/workflows/on_PR_meson_clang.yaml:15]: I'd still prefer to use the latest LTS whenever possible (i.e. both of these workflows). Older compiler are also tested w/ other CI.

Comment (neheb) [reply .github/workflows/on_PR_meson_clang.yaml:15]: Beyond the scope of this PR.

**Comment by neheb** [comment]
iconv change is actually a fix for FreeBSD. geotag for MSYS2.

**Comment by kmilos** [comment]
> iconv change is actually a fix for FreeBSD. geotag for MSYS2.

Sure, but what about https://github.com/Exiv2/exiv2/pull/2575 and https://github.com/Exiv2/exiv2/pull/2577, they look identical to me?

**Comment by neheb** [comment]
Rebased

**Review by kmilos — APPROVED** [review]

**Comment by neheb** [comment]
I'll probably do so in the future.

**Comment by kmilos** [comment]
Btw, even w/ the reduction, our CI is still running 100+ jobs and takes more than 1h.

I personally don't think that's sustainable (nor that useful) to do in the long run for every single PR.

**Pull request merged**